apps/interpreter/src/interpreter.ts

```diff
@@ -105,66 +105,112 @@ async function runPipeline(
+      blockData,
+      executionContext,
+      executionOrder,
     );
-    const inputValue =
-      parentData[0]?.value === undefined ? NONE : parentData[0]?.value;
-
-    let result: R.Result<IOTypeImplementation | null>;
-
-    // Check, if parent emitted a value
-    if (inputValue != null) {
-      try {
-        result = await blockExecutor.execute(inputValue, executionContext);
-      } catch (unexpectedError) {
-        executionContext.logger.logErrDiagnostic(
-          `An unknown error occurred: ${
-            unexpectedError instanceof Error
-              ? unexpectedError.message
-              : JSON.stringify(unexpectedError)
-          }`,
-          { node: blockData.block, property: 'name' },
-        );
-        return ExitCode.FAILURE;
-      }
-
-      if (R.isErr(result)) {
-        executionContext.logger.logErrDiagnostic(
-          result.left.message,
-          result.left.diagnostic,
-        );
-        return ExitCode.FAILURE;
-      }
-
-      blockData.value = result.right;
-
-      // If parent emitted no value, skip all downstream blocks
-    } else {
-      blockData.value = null;
-      executionContext.logger.logInfoDiagnostic(
-        `Skipped execution because parent block ${
-          parentData[0] ? parentData[0].block.name : 'NAME NOT FOUND'
-        } emitted no value.`,
+    executionContext.exitNode(block);
+
+    if (exitCode === ExitCode.FAILURE) {
+      return exitCode;
+    }
+  }
+  return ExitCode.SUCCESS;
+}
+
+async function executeBlock(
+  blockData: ExecutionOrderItem,
+  executionContext: ExecutionContext,
+  executionOrder: ExecutionOrderItem[],
+) {
+  const block = blockData.block;
+
+  const blockExecutor = createBlockExecutor(block);
+  const parentData = collectParents(block).map((parent) =>
+    executionOrder.find((blockData) => parent === blockData.block),
+  );
+  const inputValue =
+    parentData[0]?.value === undefined ? NONE : parentData[0]?.value;
+
+  let result: R.Result<IOTypeImplementation | null>;
+
+  // Check, if parent emitted a value
+  if (inputValue != null) {
+    const startTime = new Date();
+    try {
+      result = await blockExecutor.execute(inputValue, executionContext);
+      logExecutionDuration(startTime, executionContext.logger);
+    } catch (unexpectedError) {
+      executionContext.logger.logErrDiagnostic(
+        `An unknown error occurred: ${
+          unexpectedError instanceof Error
+            ? unexpectedError.message
+            : JSON.stringify(unexpectedError)
+        }`,
         { node: blockData.block, property: 'name' },
       );
+      return ExitCode.FAILURE;
     }
-    executionContext.exitNode(block);
+
+    if (R.isErr(result)) {
+      executionContext.logger.logErrDiagnostic(
+        result.left.message,
+        result.left.diagnostic,
+      );
+      return ExitCode.FAILURE;
+    }
+
+    // eslint-disable-next-line require-atomic-updates
+    blockData.value = result.right;
```

Comment:
The error occurs because the code fulfills the structure mentioned here: https://eslint.org/docs/latest/rules/require-atomic-updates#variables

I'll push another refactor to avoid this, so the error disappears and we are on the safe side.